pages/blog/the-anatomy-of-a-graphql-request.mdx

```diff
@@ -0,0 +1,341 @@
+  user(id: $id) {
+    id
+    name
+  }
+}
+```
+
+If the AST of that operation would be validated the following error will be raised.
+
+```json
+[
+  {
+    "message": "Variable \"$id\" is not defined by operation \"UserById\".",
+    "locations": [
+      {
+        "line": 2,
+        "column": 12
+      },
+      {
+        "line": 1,
+        "column": 1
+      }
+    ]
+  }
+]
+```
+
+In case any error is raised the errors are forwarded to the HTTP layer which takes care of sending them back to the client over the determined protocol. Otherwise, if no errors are raised the `execution` phase will be performed next.
+
+This procedure is performed by the `validate` function that is exported from `graphql-js`. Other languages that are orientate themselves on the `graphql-js` reference implementation have a corresponding counterpart.
+
+## GraphQL execute
+
+In the execution phase we are actually resolving the data requested by the client using the parsed and validated GraphQL operation document AST and our GraphQL schema, which contains the resolvers that specify from where the data the client requests is retrieved.
+
+Previously, the HTTP request has been parsed and normalized, which yielded the following additional (but optional) values: `variables` and `operationName`.
+
+If the GraphQL document that will be used for execution included more than one executable mutation, query or subscription operation the `operationName` is determined to identify the document that shall be used.
+In case the determined executable operation has any variable definitions, those are asserted against the `variables` values parsed from the HTTP request.
+
+If anything goes wrong or is incorrect an error is raised. E.g. the variables provided are invalid or the operation that shall be executed cannot be determined as the `operationName` is invalid or missing.
+
+**Example error for anonymous document alongside named document**
+
+```graphql
+query UserById($id: ID!) {
+  user(id: $id) {
+    id
+    name
+  }
+}
+query {
+  __typename
+}
+```
+
+```json
+{
+  "errors": [
+    {
+      "message": "This anonymous operation must be the only defined operation.",
+      "locations": [
+        {
+          "line": 7,
+          "column": 1
+        }
+      ]
+    }
+  ]
+}
+```
+
+Such an error will again be forwarded to the client by the HTTP layer.
+
+Otherwise, if no error occurs, the field values will be resolved with all the parsed and provided parameters. The phase yields a single or stream of GraphQL execution results.
+
+```json
+{
+  "data": {
+    "user": {
+      "id": "10",
+      "name": "Laurin"
+    }
+  }
+}
+```
+
+The HTTP layer forwards those to the client that initiated the request.
+
+This procedure is performed by the `execute` and `subscribe` functions that are exported from `graphql-js`. Other languages that are orientate themselves on the `graphql-js` reference implementation have a corresponding counterpart.
+
+# Why use GraphQL Helix?
```

Comment:
Arriving here, we don't know what is GraphQL Helix, and it is not explained or showcased below.

Maybe we could add a small intro to it?

Reply:
I removed this section. Let's not mention helix at all and instead focus on what envelop can do today and why yoga will be important for hooking into the HTTP phases in addition to the envelop phases.